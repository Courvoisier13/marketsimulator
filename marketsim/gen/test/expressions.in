1 + (-6*z)
-t/p -a*l
p + (if a + 12 < 9 then n - 2 else if j*u <> 7 or (j >= 0) and not o = i then 8 + j else k)
a(12, b(p, a + 8))