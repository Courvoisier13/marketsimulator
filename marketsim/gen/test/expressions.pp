1 + -6*z ->
1.0 + -6.0 * z


-t/p -a*l ->
-t / p - a * l


1 + 3 - (a + b) ->
1.0 + 3.0 - (a + b)


a*v / (a*p) ->
a * v / (a * p)


p + (if a + 12 < 9 then n - 2 else if j*u <> 7 or (j >= 0) and not o = i then 8 + j else k) ->
p + (if xxx then n - 2.0 else if xxx then 8.0 + j else k)


a(12, b(p, a + 8)) ->
QualifiedName(List(a))(List(Const(12.0), FunCall(QualifiedName(List(b)),List(Var(p), BinOp(Add(),Var(a),Const(8.0))))))


