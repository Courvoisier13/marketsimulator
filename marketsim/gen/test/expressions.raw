1 + (-6*z) ->
BinOp:
- Add
- Const:
| - 1.0
- BinOp:
| - Mul
| - Const:
| | - -6.0
| - Var:
| | - z


-t/p -a*l ->
BinOp:
- Sub
- BinOp:
| - Div
| - Neg:
| | - Var:
| | | - t
| - Var:
| | - p
- BinOp:
| - Mul
| - Var:
| | - a
| - Var:
| | - l


p + (if a + 12 < 9 then n - 2 else if j*u <> 7 or (j >= 0) and not o = i then 8 + j else k) ->
BinOp:
- Add
- Var:
| - p
- IfThenElse:
| - Less:
| | - BinOp:
| | | - Add
| | | - Var:
| | | | - a
| | | - Const:
| | | | - 12.0
| | - Const:
| | | - 9.0
| - BinOp:
| | - Sub
| | - Var:
| | | - n
| | - Const:
| | | - 2.0
| - IfThenElse:
| | - Or:
| | | - NotEqual:
| | | | - BinOp:
| | | | | - Mul
| | | | | - Var:
| | | | | | - j
| | | | | - Var:
| | | | | | - u
| | | | - Const:
| | | | | - 7.0
| | | - And:
| | | | - GreaterEqual:
| | | | | - Var:
| | | | | | - j
| | | | | - Const:
| | | | | | - 0.0
| | | | - Not:
| | | | | - Equal:
| | | | | | - Var:
| | | | | | | - o
| | | | | | - Var:
| | | | | | | - i
| | - BinOp:
| | | - Add
| | | - Const:
| | | | - 8.0
| | | - Var:
| | | | - j
| | - Var:
| | | - k


a(12, b(p, a + 8)) ->
FunCall:
- QualifiedName:
| - List:
| | - a
- List:
| - Const:
| | - 12.0
| - FunCall:
| | - QualifiedName:
| | | - List:
| | | | - b
| | - List:
| | | - Var:
| | | | - p
| | | - BinOp:
| | | | - Add
| | | | - Var:
| | | | | - a
| | | | - Const:
| | | | | - 8.0


