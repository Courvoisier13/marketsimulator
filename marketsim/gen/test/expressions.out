1 + (-6*z) ->
Add:
- Const:
| - 1.0
- Mul:
| - Const:
| | - -6.0
| - Var:
| | - z


-t/p -a*l ->
Sub:
- Div:
| - Neg:
| | - Var:
| | | - t
| - Var:
| | - p
- Mul:
| - Var:
| | - a
| - Var:
| | - l


p + (if a + 12 < 9 then n - 2 else if j*u <> 7 or (j >= 0) and not o = i then 8 + j else k) ->
Add:
- Var:
| - p
- IfThenElse:
| - Less:
| | - Add:
| | | - Var:
| | | | - a
| | | - Const:
| | | | - 12.0
| | - Const:
| | | - 9.0
| - Sub:
| | - Var:
| | | - n
| | - Const:
| | | - 2.0
| - IfThenElse:
| | - Or:
| | | - NotEqual:
| | | | - Mul:
| | | | | - Var:
| | | | | | - j
| | | | | - Var:
| | | | | | - u
| | | | - Const:
| | | | | - 7.0
| | | - And:
| | | | - GreaterEqual:
| | | | | - Var:
| | | | | | - j
| | | | | - Const:
| | | | | | - 0.0
| | | | - Not:
| | | | | - Equal:
| | | | | | - Var:
| | | | | | | - o
| | | | | | - Var:
| | | | | | | - i
| | - Add:
| | | - Const:
| | | | - 8.0
| | | - Var:
| | | | - j
| | - Var:
| | | - k


a(12, b(p, a + 8)) ->
FunCall:
- QualifiedName:
| - List:
| | - a
- List:
| - Const:
| | - 12.0
| - FunCall:
| | - QualifiedName:
| | | - List:
| | | | - b
| | - List:
| | | - Var:
| | | | - p
| | | - Add:
| | | | - Var:
| | | | | - a
| | | | - Const:
| | | | | - 8.0


